editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -1061,132 +1061,133 @@ void CanvasItemEditor::_switch_theme_preview(int p_mode) {
 	EditorNode::get_singleton()->update_preview_themes(theme_preview);
 }
 
+// Updates dragged guide index, drag start position, and drag type in response to mouse motion and left clicks over guides.
+void CanvasItemEditor::_is_hovering_guide(const Point2 &p_pos, bool p_is_pressed) {
+	Node *scene = EditorNode::get_singleton()->get_edited_scene();
+	Array vguides = scene->get_meta("_edit_vertical_guides_", Array());
+	Array hguides = scene->get_meta("_edit_horizontal_guides_", Array());
+	Transform2D xform = viewport_scrollable->get_transform() * transform;
+	int hovered_guide_index = -1;
+
+	if (p_pos.x < RULER_WIDTH) {
+		// Check if we are hovering an existing horizontal guide.
+		for (int i = 0; i < hguides.size(); i++) {
+			if (ABS(xform.xform(Point2(0, hguides[i])).y - p_pos.y) < 8) {
+				is_hovering_h_guide = true;
+				hovered_guide_index = i;
+				break;
+			}
+		}
+		if (!is_hovering_h_guide && p_is_pressed) {
+			drag_type = DRAG_V_GUIDE;
+		}
+	} else if (p_pos.y < RULER_WIDTH) {
+		// Check if we are hovering an existing vertical guide.
+		for (int i = 0; i < vguides.size(); i++) {
+			if (Math::abs(xform.xform(Point2(vguides[i], 0)).x - p_pos.x) < 8) {
+				is_hovering_v_guide = true;
+				hovered_guide_index = i;
+				break;
+			}
+		}
+		if (p_is_pressed) {
+			drag_from = xform.affine_inverse().xform(p_pos);
+			if (!is_hovering_v_guide) {
+				drag_type = DRAG_H_GUIDE;
+			}
+		}
+	}
+	if (p_is_pressed) {
+		dragged_guide_index = hovered_guide_index;
+		if (hovered_guide_index >= 0) {
+			drag_type = is_hovering_h_guide ? DRAG_H_GUIDE : DRAG_V_GUIDE;
```

Comment:
This assignment duplicates the one above. No need to set `drag_type` in two places...